xbmc/Application.cpp

```diff
@@ -1551,8 +1550,11 @@ void CApplication::ReloadSkin(bool confirm/*=false*/)
        user as to whether they want to keep the current skin. */
     if (confirm && !m_skinReverting)
     {
-      bool cancelled;
-      if (!CGUIDialogYesNo::ShowAndGetInput(CVariant{13123}, CVariant{13111}, cancelled, CVariant{""}, CVariant{""}, 10000))
+      CVariant options{CVariant::VariantTypeObject};
+      options["heading"] = 13123;
+      options["text"] = 13111;
+      options["autoclose"] = 10000; //timeout value for autoclosing the dialog
+      if (CApplicationMessenger::Get().SendMsg(TMSG_GUI_DIALOG_YESNO, -1, -1, static_cast<void*>(&options)) != 1)
```

Comment:
IMO this bloats the code massivly, why not introduce a helper class + method and do these stuff there?